frontend/model/contracts/shared/functions.js

```diff
@@ -156,3 +156,30 @@ export function makeMentionFromUsername (username: string): {
     all: '@all'
   }
 }
+
+// group.js, mailbox.js, chatroom.js related
+
+function setContractJoining (contractID: string, finishedJoining: boolean) {
+  const joiningContracts = sbp('okTurtles.data/get', 'JOINING_CONTRACTS') || {}
+  if (finishedJoining) {
+    delete joiningContracts[contractID]
+  } else {
+    joiningContracts[contractID] = true
+  }
+  sbp('okTurtles.data/set', 'JOINING_CONTRACTS', joiningContracts)
+}
+
+export async function syncContract (contractID: string, logJoining: boolean = false) {
+  if (logJoining) {
+    setContractJoining(contractID, false)
+  }
+  await sbp('chelonia/contract/sync', contractID)
+  if (logJoining) {
+    setContractJoining(contractID, true)
+  }
+}
```

Comment:
What are these functions?

Why did you create them?

Can you use `'chelonia/contract/isSyncing'` instead?

Reply:
These functions are needed to resolve #1553.
When a user tries to log in, he tries to sync all the contracts no matter it's the first login in a new device or the contracts were already fully/half synced. We need to separate these two and `'chelonia/contract/isSyncing'` is not enough for this since those two are all syncing.
And also users sync the chatroom contract when they join it and we need to differentiate this too.

I differentiate those by calling the `syncContract` function with different value of parameter `logJoining` and check them by calling `checkContractJoining` function in `model/contracts/chatroom.js`.

Reply:
I will add comments to these functions.

Reply:
I have just got an idea to create `'chelonia/contract/isNewSyncing'` instead of these functions.